sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/SensorContextUtilsTest.java

```diff
@@ -157,11 +161,108 @@ public void hasFilesOfLanguage_whenOnlyOtherLanguageExists_returnsFalse() {
     assertThat(hasFilesOfLanguage(fs, LANG_ONE)).isFalse();
   }
 
+  @Test
+  public void toTextRange_whenMultiLineRangeStartsAtEOL_doesNotFilterOut() {
+    var inputFile = new TestInputFileBuilder("mod", "source.cs")
+      .setLanguage("cs")
+      .setType(Type.MAIN)
+      .setContents(
+        "Some text \n" +
+        "rangeStartingAtEOL1\n" +
+        "Some more text rangeStarting\n" +
+        "AtEOL2 Some other text")
+      .build();
+    fs.add(inputFile);
+    assertTextRange(toTextRange(inputFile, pbTextRangeOf(1, 10, 2, 18)), 1, 10, 2, 18);
+    assertTextRange(toTextRange(inputFile, pbTextRangeOf(2, 19, 4, 5)), 2, 19, 4, 5);
+  }
+
+  @Test
+  public void toTextRange_whenMultiLineRangeStartsBeyondEOL_trimsStartMovingToNextLine() {
+    var inputFile = new TestInputFileBuilder("mod", "source.cs")
+      .setLanguage("cs")
+      .setType(Type.MAIN)
+      .setContents("\tSome text\nrangeStartingAtEOL1\nrangeStarting\nAtEOL2\nAndSpanning\nMultipleLines")
+      .build();
+    fs.add(inputFile);
+
+    // Possible real scenario: 4 spaces transformed into \t -> new EOL at 10 instead of 13
+    assertTextRange(toTextRange(inputFile, pbTextRangeOf(1, 13, 2, 18)), 2, 0, 2, 18);
+  }
+
+  @Test
+  public void toTextRange_whenMultiLineRangeEndsBeyondEOL_trimsBasedOnEndLineLength() {
+    var inputFile = new TestInputFileBuilder("mod", "source.cs")
+      .setLanguage("cs")
+      .setType(Type.MAIN)
+      .setContents("Some text multiline\nRangeWithEndLineOffsetBiggerThanStartLineOffset Some other text")
+      .build();
+    fs.add(inputFile);
+    assertTextRange(toTextRange(inputFile, pbTextRangeOf(1, 10, 2, 46)), 1, 10, 2, 46);
```

Comment:
I think the provided range does not match the test title. The second line has a length of 64 that is longer than 46 and doesn't need to be trimmed.

Reply:
As discussed, I changed the test content to truly reflect the name of the test.